paddle/framework/operator.h

```diff
@@ -319,5 +319,14 @@ class OperatorWithKernel : public OperatorBase {
   virtual void InferShape(const InferShapeContext& ctx) const = 0;
 };
 
+namespace op_helpers {
+
+// Generate the gradient variable's name of a forward varialbe
+inline std::string GenGradName(const std::string& var) {
```

Comment:
good ideas.  but there are some strange `GRAD_VAR_SUFFIX` usages, such as

https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/backward.cc#L138
https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/backward.cc#L150

It seems that `GRAD_VAR_SUFFIX` is necessary for these usages,  `GenGradName` is added to hide the details of grad var's name from developers.
@Canpio 

The namespace `op_helpers` is added, because operators needs many helper functions, `GenGradName` is the first, many other ones should be offered, such some helpers to help these simple check:

```c++
PADDLE_ENFORCE(ctx.InputSize() == 2,
               "Input size of OnehotCrossEntropyOp must be two");
PADDLE_ENFORCE(ctx.OutputSize() == 1,
               "Output size of OnehotCrossEntropyOp must be one");
```
may need a helper
```
void EnforceInputsOutputsCountMatchProto();
```

```c++
PADDLE_ENFORCE(ctx.InputVar(0) != nullptr && ctx.InputVar(1) != nullptr,
               "Inputs of OnehotCrossEntropyOp must all be set");
PADDLE_ENFORCE(ctx.OutputVar(0) != nullptr,
               "Outputs of OnehotCrossEntropyOp must all be set");
```
may need a helper

```
void EnforceVarsValid(xxx);
```

```c++
PADDLE_ENFORCE(ctx.Input<Tensor>(0)->dims() == ctx.Input<Tensor>(1)->dims(),
               "Two input of Add Op's dimension must be same.");
```
may need a helper
```c++
void EnforceShapeMatches(const std::vector<std::string>& var_names);
```

So, I will create an `op_helpers` directory and keep this namespace.
@wangkuiyi